### ISSUE TITLE
COCO import may take longer than 30 gunicorn default timeout

### ISSUE THREAD

**Issue by matt-deboer** [jsbroks/coco-annotator#87]
I've noticed when importing relatively large COCO files (about 3k images with 4,5 annotations per image) that the process can take longer than 30 seconds, causing the client to be disconnected.

I was able to fix it by just increasing the default gunicorn timeout to 90 seconds.

If a 90 second default is not acceptable, maybe the annotations upload could return immediately and provide a callback URL (API method) that the user could check to see if the import is one of "in-progress", "failed", or "successful"?

**Comment by jsbroks** [comment]
Good idea, I'm also looking into adding socket support for more realtime.

**Comment by matt-deboer** [comment]
With a little more testing, I'm finding that this process definitely takes too long on a reasonably-sized dataset; I'm timing out even with 180 seconds to complete. It looks like the process could benefit from some parallelization and/or batching.
